<a href="https://colab.research.google.com/github/khing-fm/Data_Science/blob/main/pandas_for_data_analysi.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [5]:
# Import pandas Iconvention: import as 'pd')
import pandas as pd
import numpy as np

print(f"pandas version: {pd.__version__}")
print(f"numpy version: {np.__version__}")

pandas version: 2.2.2
numpy version: 2.0.2


In [7]:
np.arange(0, 15, 2)

array([ 0,  2,  4,  6,  8, 10, 12, 14])

In [9]:
# Example: Creating a Series

# From a list
sales = pd.Series([100, 150, 200, 175, 225])
print("series from list:")
print(sales)
print(sales.index)
print(f"\nIndex: {sales.index.tolist()}")
print(f"Values: {sales.values.tolist()}")

series from list:
0    100
1    150
2    200
3    175
4    225
dtype: int64
RangeIndex(start=0, stop=5, step=1)

Index: [0, 1, 2, 3, 4]
Values: [100, 150, 200, 175, 225]


In [10]:
sales_mean = sales.mean()
sales_std = sales.std()

print(f"Mean Sales: {sales_mean}")
print(f"Standard Deviation of Sales: {sales_std}")

Mean Sales: 170.0
Standard Deviation of Sales: 48.088460154178364


### Sales Data Analysis Summary

*   **Mean Sales (170.0)**: Represents the average value of the sales entries.
*   **Standard Deviation (~48.09)**: Indicates the spread of the data; a higher value suggests that the sales figures vary significantly from the average.

In [14]:
pd.Series(np.array([10, 20, 30, 40, 50]), dtype="int16")

,0
0,10
1,20
2,30
3,40
4,50


### Creating a Series with NumPy and dtypes

This example demonstrates:
1.  **NumPy Integration**: Creating a Series directly from an `np.array`.
2.  **Memory Optimization**: Using `dtype="int16"` to store integers using less memory (16 bits) compared to the standard 64-bit default.